### PR TITLE
Curator search has default value if not set, and if searched for outs…

### PR DESCRIPTION
**Pull request by sondrev**
…ide of allowed scope will return error

### CODE REVIEW

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/utils/GraphUtils.java [hunk 1]
```diff
@@ -19,6 +19,9 @@
 import org.slf4j.LoggerFactory;
 
 public class GraphUtils {
+    public static final String APPLICATION_JSON = "application/json";
+    public static final String PART_OF_PROPERTY = "https://nva.sikt.no/ontology/publication#partOf";
+    public static final String HAS_PART_PROPERTY = "https://bibsysdev.github.io/src/organization-ontology.ttl#hasPart";
```
Comment (sondrev) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/utils/GraphUtils.java:24]: @brinxmat We need to fix this

Comment (sondrev) [reply nvi-commons/src/main/java/no/sikt/nva/nvi/common/utils/GraphUtils.java:24]: We should change the datamodel so we can use https://nva.sikt.no/ontology/publication#hasPart

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java [hunk 2]
```diff
@@ -49,20 +68,45 @@ protected PaginatedSearchResult<NviCandidateIndexDocument> processInput(Void inp
...
         var username = requestInfo.getUserName();
         var year = extractQueryParamPublicationDateOrDefault(requestInfo);
 
+        assertUserIsAllowedToSearchAffiliations(affiliations, topLevelOrg);
+
         var candidateSearchParameters = new CandidateSearchParameters(affiliations, excludeSubUnits, filter, username,
-                                                                      year, customer, offset, size);
+                                                                      year, topLevelOrg, offset, size);
 
         return attempt(() -> openSearchClient.search(candidateSearchParameters))
```
Comment (truhacevkir) [inline index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java:83]: I think we will soon need a method that extracts all searchparams from requestInfo, something like:

`CandidateSearchParams.fromRequestInfo(requestInfo)`

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/OpenSearchClientTest.java [hunk 3]
```diff
@@ -384,10 +391,12 @@ private static PublicationDetails randomPublicationDetails() {
                                       List.of());
     }
 
-    private static PublicationDetails randomPublicationDetailsWithYear(String year) {
+    private static PublicationDetails randomPublicationDetailsWithYearAndCurator(String year, String affiliation) {
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/OpenSearchClientTest.java:394]: `randomPublicationDetailsWithYearAndContributor`

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/OpenSearchClientTest.java [hunk 4]
```diff
@@ -138,11 +143,11 @@ void shouldReturnDocumentsFromIndexAccordingToGivenOffsetAndSize() throws IOExce
 
     @Test
     void shouldReturnEverythingIfYearFilterAndAffiliationsAreNotSet() throws IOException, InterruptedException {
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/OpenSearchClientTest.java:145]: Confused by the test name and searchParameters in the test. The test is called `shouldReturnEverythingIfYearFilterAndAffiliationsAreNotSet`, but you are adding affiliations as search params in the test?
Rename test?

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java [hunk 5]
```diff
@@ -3,10 +3,11 @@
 import static no.sikt.nva.nvi.index.SearchNviCandidatesHandler.QUERY_PARAM_EXCLUDE_SUB_UNITS;
```
Comment (monaullah) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandlerTest.java:3]: Add another test `shouldReturnPaginatedSearchResultWithUserTopLevelOrgAsDefaultAffiliationIfNotSet()`?

**Pull request merged**